### PR TITLE
Refactor to use op graph + fixes; inputs/output

### PR DESCRIPTION
**Pull request by l-hoang**
Renames querytest to graph-query and changes it to use the OpGraph. A
hack was added to deal with distributed count for the time being as the
tables are not synchronized. Some new functions were added to
QueryServer and Table to facilitate this change.
 
Updates the test inputs to v4: some queries had to be changed binding
the variable from a path to a vertex because the OpGraph (and
technically even the older execution engine) did not support return
variables bound to paths, and this was causing tests to fail. v4 has
removed the queries and verify file: instead, they have been moved to
the graph-query directory so that a small change does not require a
completely new inputs tar to be created.
 
Also, a rather important bug was fixed in Table: column index and value
were swapped and can cause crashes.

Depends on #419 as it builds on top of it.

### CODE REVIEW

File: libquery/src/QueryServer.cpp [hunk 1]
```diff
@@ -1,32 +1,52 @@
...
+galois::graphs::AttributedGraph*
+galois::QueryServer::LoadPropertyGraph(std::string filename) {
   // try to find graph in graphs_
   auto found_graph = graphs_.find(filename);
   if (found_graph != graphs_.end()) {
+    GALOIS_LOG_DEBUG("Graph {} already loaded by the server", filename);
     // graph exists, return it
     return found_graph->second.get();
   } else {
+    GALOIS_LOG_DEBUG("Loading graph {}", filename);
```
Comment (roshandathathri) [inline libquery/src/QueryServer.cpp:49]: VERBOSE is better suited for both the changes above.

File: libquery/src/Table.cpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ galois::Table::Builder galois::Table::Builder::From(
                          data_src_id, data_dst_id);
       }
       uint64_t edge_id = *(found_edge.value());
-      builder.AddUInt64(edge_id, col);
+      builder.AddUInt64(col, edge_id);
```
Comment (roshandathathri) [inline libquery/src/Table.cpp:176]: I would prefer the bug-fix to be a separate commit when you rebase and merge.

File: lonestar/querying/cpu/graph-query/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    endif()
+
+    add_test(NAME run-dist-graph-query_${np} COMMAND mpiexec --bind-to none -n ${np} $<TARGET_FILE:graph-query> -queryBasePath=${samples} -listOfQueries=${samples}/${query_input} ${ldbc_input} -t=${thr} -output -outputLocation=${output_location})
+    add_test(NAME verify-dist-graph-query_${np} COMMAND python ${RESULT_CHECKER} -delete=1 -stringcolumn=1 ${query_output} ${output_location}/queries_p${np}_t${thr}.count)
+
+    # add fixtures to prevent verify from running before run
+    set_tests_properties(run-dist-graph-query_${np}
+      PROPERTIES
+        ENVIRONMENT GALOIS_DO_NOT_BIND_THREADS=1
+        FIXTURES_SETUP run-dist-graph-query_${np})
```
Comment (roshandathathri) [inline lonestar/querying/cpu/graph-query/CMakeLists.txt:69]: When you merge conflicts, please make sure that you incorporate changes from the current querytest CMakeLists.txt w.r.t environment variable ASAN_OPTIONS

File: lonestar/querying/cpu/graph-query/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,77 @@
+add_executable(graph-query graph-query.cpp)
+target_link_libraries(graph-query galois_query lonestar)
+
+set(query_input querylist.txt)
+set(samples ${CMAKE_CURRENT_SOURCE_DIR}/cypherqueries/)
+set(ldbc_input ${BASEINPUT}/propertygraphs/ldbc_003/meta)
+set(query_output ${CMAKE_CURRENT_SOURCE_DIR}/cypherqueries/003_${query_input}.truth)
```
Comment (roshandathathri) [inline lonestar/querying/cpu/graph-query/CMakeLists.txt:7]: Isn't the notation to use cypher-queries directory?

File: lonestar/querying/cpu/graph-query/graph-query.cpp [hunk 5]
```diff
@@ -108,52 +111,27 @@ size_t ProcessQueryFile(
...
-  galois::StatTimer graph_construct_time("GraphConstructTime");
-  graph_construct_time.start();
-  galois::graphs::AttributedGraph att_graph;
-  // default load is katana form; give it a meta file specifying how to load
-  // it in
-  galois::LoadKatanaFormToCypherGraph(&att_graph, filename);
-  graph_construct_time.stop();
   if (host_id == 0) {
-    att_graph.ReportGraphStats();
+    galois::gInfo("Num matched subgraphs ", num_match);
```
Comment (roshandathathri) [inline lonestar/querying/cpu/graph-query/graph-query.cpp:129]: This comment won't make sense for all queries going forward. Technically, it's "Num rows in the output ".

File: lonestar/querying/cpu/graph-query/graph-query.cpp [hunk 6]
```diff
@@ -195,6 +173,38 @@ int main(int argc, char** argv) {
...
+  galois::QueryServer query_server;
+
+  // preload the graph
+  galois::StatTimer graph_construct_time("GraphConstructTime");
+  graph_construct_time.start();
+  query_server.PreloadGraph(filename);
+  graph_construct_time.stop();
+  // holds parameter values for cypher query passed in via command line
+  std::unordered_map<std::string, std::string> parameter_map =
+      ParseParameters(host_id);
```
Comment (roshandathathri) [inline lonestar/querying/cpu/graph-query/graph-query.cpp:207]: So we can support parameters? When are parameters substituted?

@bowu As we are not saving the execution plan, we can substitute them before creating the execution plan, right?

File: lonestar/querying/cpu/graph-query/CMakeLists.txt (hunk 3 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/CMakeLists.txt:69]: yeah i should have realized that the rebase was complaining that I deleted the CMake for a reason

File: lonestar/querying/cpu/graph-query/CMakeLists.txt (hunk 4 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/CMakeLists.txt:7]: do you mean that cypherqueries should be called cypher-queries?

File: lonestar/querying/cpu/graph-query/graph-query.cpp (hunk 5 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/graph-query.cpp:129]: yup, will fix

File: lonestar/querying/cpu/graph-query/graph-query.cpp (hunk 6 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/graph-query.cpp:207]: it's been a while since I added them in, so I have to check; will get back to you in a bit

File: lonestar/querying/cpu/graph-query/CMakeLists.txt (hunk 4 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/CMakeLists.txt]: went ahead and changed the name, let me know if that's not what you meant

File: lonestar/querying/cpu/graph-query/graph-query.cpp (hunk 6 above)
Comment (l-hoang) [reply lonestar/querying/cpu/graph-query/graph-query.cpp:207]: update: parameter support is currently very limited; it only works for skip/limit

File: lonestar/querying/cpu/graph-query/CMakeLists.txt (hunk 4 above)
Comment (roshandathathri) [reply lonestar/querying/cpu/graph-query/CMakeLists.txt]: Yes, directory name. Sorry for not being clear.

File: lonestar/querying/cpu/graph-query/graph-query.cpp (hunk 6 above)
Comment (roshandathathri) [reply lonestar/querying/cpu/graph-query/graph-query.cpp:207]: Ok, thanks for the clarification
